### ISSUE TITLE
Why there is no LlavaForSequenceClassification ?

### ISSUE THREAD

**Issue by Zuhashaik** [huggingface/transformers#31814]
### Feature request

If there is a LlamaForSequenceClassification class we can use decoder only models for classification tasks.

### Motivation

If there is a LlamaForSequenceClassification class we can use decoder only models for classification tasks.decoder-only

### Your contribution

Allow me to contribute to this as I am already working on this model for mental health meme classification for the upcoming AAAI conference.

**Comment by amyeroberts** [comment]
Hi @Zuhashaik, thanks for opening this issue! 

We only add specific task heads for models if there are official checkpoints available; the task is described in the original paper or there's a large community demand.

**Comment by NielsRogge** [comment]
Note that you could technically also use `LlavaForConditionalGeneration` by training the model to only spit out one classification token per image.

**Comment by Zuhashaik** [comment]
> Note that you could technically also use `LlavaForConditionalGeneration` by training the model to only spit out one classification token per image.

Yeah I've attached an score layer at the top of the model by passing those last layer hidden states. While doing this, some cuda related issues ("data in different devices cuda:0 and cuda:1") came up.

So I've changed my `LlavaForConditionalGeneration` class itself in modelling_llava.py file (added score layer) and used that for classification.

Thankyou.

**Comment by Zuhashaik** [comment]
I am facing lots of data type and quantization errors while defining classifier (score) separately please initiate `LlavaForSequenceClassification` and SequenceClassification for vision models..

> Hi @Zuhashaik, thanks for opening this issue!
> 
> We only add specific task heads for models if there are official checkpoints available; the task is described in the original paper or there's a large community demand.

Thank you!

**Comment by uebian** [comment]
I'm also looking forward to `LLavaForSequenceClassification`

**Comment by Zuhashaik** [comment]
Facing lots of issues without this class, @amyeroberts please do the needful thank-you

**Comment by Jintao-Huang** [comment]
+1

**Comment by Zuhashaik** [comment]
> +1

@NielsRogge Please do the needful. It will be very much handy dealing with sequence cls class rather than using conditional generation class.

Thank you

**Comment by Shaik1903** [comment]
@NielsRogge Please add the class
